ddtrace/tracer/textmap.go

```diff
@@ -349,6 +349,8 @@ func (p *propagator) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapWr
 	if !ok || ctx.traceID.Empty() || ctx.spanID == 0 {
 		return ErrInvalidSpanContext
 	}
+	ctx.mu.Lock()
+	defer ctx.mu.Unlock()
```

Comment:
This is just to protect `ctx.baggage`, right? In that case, we should just use `(c *spanContext) ForeachBaggageItem` which handles the lock for us. That will also help us hold the lock for as little time as possible.

Reply:
I went ahead and pushed that change.